### PR TITLE
[AMORO-3118][Improvement]: Move iceberg codes from amoro-core to a new module amoro-iceberg

### LINKED ISSUE #3118: [Subtask]: Decouple Amoro Core from Iceberg to achieve equal support for different table formats
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Amoro Core module will no longer depend on Iceberg module and no longer include mixed-format related logic.

Amoro Core module will be used to define how to integrate with amoro across different formats.

### Parent issue

#3116

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

Currently, there are too many logics related to mixed-format in our amoro-core module. 

In order to make amoro to support multiple formats better, amoro needs to have the ability to run without Iceberg. 

Therefore, I moved the codes related to mixed-format in `amoro-core` to the `mixed-format` module. 

In the future, the `amoro-core` module should focus on the interface definitions across formats.

close #3118

## Brief change log

I moved all the codes related to mixed-format in `amoro-core` to a new module `amoro-mixed-format/amoro-mixed-format-core`, and only retained some common codes in core, such as those related to Client and UnifiedCatalog.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-core/src/main/java/org/apache/amoro/utils/MixedCatalogUtil.java [hunk 1]
```diff
@@ -21,28 +21,12 @@
 import org.apache.amoro.TableFormat;
 import org.apache.amoro.api.CatalogMeta;
 import org.apache.amoro.api.TableMeta;
-import org.apache.amoro.io.AuthenticatedFileIO;
```
Comment (zhoujinsong) [inline amoro-core/src/main/java/org/apache/amoro/utils/MixedCatalogUtil.java:24]: It seems we should rename this class to `CatalogUtil` now.

**Review by zhoujinsong — COMMENTED** [review]
I left some minor suggestions, PTAL. @baiyangtx

File: amoro-ams/amoro-ams-api/src/main/java/org/apache/amoro/api/config/DataExpirationConfig.java [hunk 2]
```diff
@@ -166,9 +166,7 @@ public DataExpirationConfig(MixedTable table) {
   }
 
   public static DataExpirationConfig parse(Map<String, String> properties) {
-    boolean gcEnabled =
-        CompatiblePropertyUtil.propertyAsBoolean(
-            properties, org.apache.iceberg.TableProperties.GC_ENABLED, true);
+    boolean gcEnabled = CompatiblePropertyUtil.propertyAsBoolean(properties, "gc.enabled", true);
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply amoro-ams/amoro-ams-api/src/main/java/org/apache/amoro/api/config/DataExpirationConfig.java]: fixed

File: amoro-core/src/main/java/org/apache/amoro/utils/MixedCatalogUtil.java (hunk 1 above)
Comment (baiyangtx) [reply amoro-core/src/main/java/org/apache/amoro/utils/MixedCatalogUtil.java]: fixed

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**